core/domain/topic_fetchers.py

```diff
@@ -747,3 +748,31 @@ def get_canonical_story_dicts(
         canonical_story_dicts.append(story_summary_dict)
 
     return canonical_story_dicts
+
+
+def get_published_story_exploration_ids(
+    topic_id: Optional[str] = None
+) -> List[str]:
+    """Returns a list of all exploration ids belonging to each story of either
+    a topic with id topic_id if provided or all topics.
+
+    Args:
+        topic_id: str|None. The id of the topic to search through. When not
+            provided, all topics are searched through.
+
+    Returns:
+        list(str). A list of all exploration ids belonging to the stories in the
+        topic(s) to search through.
+    """
+    mappings = ([
+        summary.published_story_exploration_mapping
+        for summary in (
+            [get_topic_summary_by_id(topic_id)]
```

Comment:
@seanlip The challenge I'm having with `get_multi` is that it requires a list of topic ids to be provided. But when the topic name is not given by the client, the app is supposed to fetch all topics, which you said is too slow -- O(N) queries.

Maybe rather than running a query to fetch each TopicSummaryModel entity individually, I could run a query to fetch each TopicSummaryModel *key* individually (which is probably faster than the former). Since the key of `TopicSummaryModel` is just the topic id, I could then provide the keys as an argument to `get_multi`. Idk if its faster though

Reply:
Actually, sorry @StephenYu2018, I completely misread this -- it is fine as you have it. (I thought we were fetching multiple topic IDs one at a time but I missed that that was only in the "if topic_id is provided" case.) Sorry about that, please ignore my original comment here!